terraform/cluster/aws/main.tf

```diff
@@ -105,7 +106,7 @@ resource "aws_eks_node_group" "cluster" {
   scaling_config {
     desired_size = 1
     min_size     = 1
-    max_size     = 100
+    max_size     = var.high_availability ? 100 : 3
```

Comment:
Leaving it at 3 when disabled as depending on the workloads it could run out of NICs for pods(t3 allows 12 per instance and rack + k8s/addon resources already get 13)